crates/sui-types/src/messages.rs

```diff
@@ -1931,13 +1945,13 @@ impl TransactionDataAPI for TransactionDataV1 {
                 SingleTransactionKind::Call(_)
                 | SingleTransactionKind::TransferObject(_)
                 | SingleTransactionKind::Pay(_)
-                | SingleTransactionKind::Publish(_) => true,
+                | SingleTransactionKind::Publish(_)
+                | SingleTransactionKind::ProgrammableTransaction(_) => true,
```

Comment:
This effectively makes pay/pay all/transfer sui sponsorable via programmable transactions.  Why were they not sponsorable before, was it a technical limitation we wanted to fix later, or a moral objection to sponsoring these kinds of transactions? cc @longbowlu

Reply:
Good question! I had spoke to @longbowlu about this, this is the summary: 
I believe Pay* were banned for a moral objection? In that they are sort of meaningless in the sponsored case since the spender isn't doing anything. 
For programmable transactions, the sponsor can look to see if they approve of the way the `Argument::GasCoin` is being used. I think it opens up some neat doors too, since now they can pay for additional things besides gas, if they so choose.

Reply:
The hypothesis is that sponsor is only in the business of paying the gas of a transaction In the case of *sui transactions, the gas would be instead transferred which is against the motivation.
With programmable transactions, since everything is now consolidated, it would be up to the sponsor to examine if they want to sponsor the requested transaction.